### ISSUE TITLE
SlackFormatter alternative - just a function

### ISSUE THREAD

**Issue by aoberoi** [slackapi/hubot-slack#456]
### Description

the `SlackFormatter` class's functionality today is actually just one method: `incoming()`. i believe its job is better done by a simple function.

the first problem is that it contains a reference to the RtmClient#dataStore, which is being deprecated (See #385). it uses the dataStore to replace links, mentions, and other special formatting with more plain representations in a string (such as `"@username"` instead of `"<@U123456|username>"`). to do so, it sometimes relies on methods to translate a user ID or a channel ID into user names or channel names, respectively. an alternative is to use the Web API methods `users.info` and `channels.info` to fetch that just in time. another idea is to use the robot brain to store and fetch this data.

next, the function actually removes valuable information from the message. we should make this substitution optional, where the default is currently set to preserve behavior (and not create API breaking changes). even with the substitution occurring, we should augment the message with properties that reference each of the otherwise lost IDs. here is an illustration of this idea:

```
message = {
  text: 'hello @bob, let's take this to the #parkinglot',
  mentions: {
    'bob': 'U123456'
  },
  channelLinks: {
    'parkinglot': 'C98765'
  }
};
```

we can leave the class in place, and even instantiate it for the `SlackClient`, until the next major version release. the code and documentation should clearly mark it as deprecated, and should be removed at that time.

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/hubot-slack/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

**Comment by aoberoi** [comment]
It turns out that the `options.returnRawText` option effectively adds a new `rawText` property to `SlackTextMessage` for the loss-less message text.

**Comment by aoberoi** [comment]
related: #133 looked really good and somehow was lost to history.

**Comment by aoberoi** [comment]
#461 is working towards fixing this.

**Comment by aoberoi** [comment]
will be fixed by #465
